### PR TITLE
Replace TravisCI with Github Actions

### PR DESCRIPTION
**Pull request by Christopher-Tom**

### CODE REVIEW

**Review by hsorathia — APPROVED** [review]

**Review by seemavora — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Discord Bot Tests
+      
+on:
```
Comment (jerrylee17) [inline .github/workflows/main.yml:3]: Should probably also do `on: push` so its run every time you push

**Review by jerrylee17 — APPROVED** [review]
A few questions:

1. Does this automatically run `npm install` (not that it really matters that much)?
2. Are there any limitations to how many times you can run CI? For ex: Travis-ci gives us a set amount of points so you cannot keep running it

File: .github/workflows/main.yml (hunk 1 above)
Comment (evanugarte) [reply .github/workflows/main.yml:3]: resolving this as we decided to just do prs in discord
image

**Comment by evanugarte** [comment]
@jerrylee17 
1. yes
2. yes, but i doubt we will exceed it

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  pull_request:
+    branches: [ master ]
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:5]: I think we could change this to run on all PRs, not just for master.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:16]: I don't think `matrix` is defined? If we don't specify `with`, will it use get the right version of node?

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
I requested some changes.
btw, i was looking at [this](https://docs.github.com/en/actions/guides/building-and-testing-nodejs, [this](https://github.com/actions/setup-node/), and [this](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software) for reference on GitHub actions. I'm not really sure if what I understand is accurate.

File: .github/workflows/main.yml (hunk 3 above)
Comment (charlotte-zhuang) [reply .github/workflows/main.yml:16]: Actually, maybe it's best to specify `matrix.node-version` with `14.x` to match `package.json`.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+name: Discord Bot Tests
+      
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:9]: `runs-on: ubuntu-18.04` would match our production env.

**Review by ThaiQ — APPROVED** [review]

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+       matrix:
+         os: [macos-latest, windows-latest, ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (evanugarte) [inline .github/workflows/main.yml:18]:
```suggestion
          node-version: 14.x
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+       matrix:
+         os: [macos-latest, windows-latest, ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
```
Comment (evanugarte) [inline .github/workflows/main.yml:15]:
```suggestion
      - name: Use Node.js 14.x
```

File: .github/workflows/main.yml (hunk 2 above)
Comment (evanugarte) [reply .github/workflows/main.yml:5]: this means run the test when we open a pr for master, this pr is an example

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+name: Discord Bot Tests
+      
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
```
Comment (charlotte-zhuang) [inline .github/workflows/main.yml:9]: Any reason why we're using a matrix?

I think we should change it to `runs-on: ubuntu-18.04`

File: .github/workflows/main.yml (hunk 2 above)
Comment (charlotte-zhuang) [reply .github/workflows/main.yml:5]: bump

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
Requested changes to OS version primarily. Also wondering about `pull_request:`

File: .github/workflows/main.yml (hunk 4 above)
Comment (charlotte-zhuang) [reply .github/workflows/main.yml]: new comment on the same code

**Review by charlotte-zhuang — APPROVED** [review]

**Pull request merged**